### DISCUSSION TITLE
Fetching data from the year specified dynamically

### DISCUSSION THREAD

**Discussion by jacquesvdhorst** [supabase/supabase#11526] [Questions] [answered]
So I've been struggling with this filter / range for my Supabase query. Currently I'm using NextJs 13 and using some dynamic routing with links.

The user clicks on the year and then the quarter of the year they want to view. Then instead of having hard coded data fetches I'd like to fetch all the data from that specific year. 

In my database I've got a table called matches for example and inside of that I have a column called date which has the date object inside: 
![image](https://user-images.githubusercontent.com/35291167/211205020-3272b59c-ca82-4445-a12c-de2c6c4d9b01.png)

Using the params.year objects I can get the year as a string - lets say '2022' - Is it possible to adjust my fetch query to use that params.year to set it for all rows having the year 2022 in it?

Or is it the case of I need to add in a function to take the params.year and convert it to a year object from start of 2022 to end and then use that as a .gte() and .lte() so its not as hard coded?

Any help in a direction or solution would be greatly appreciated ♥

**Comment by GaryAustin1** [comment, marked as answer]
The gte and lte equal approach should work.

**Reply by jacquesvdhorst** [reply]
Alright yeah so I've opted to great a function to help with the gte and lte filters. If anyone needs help this is what I did:

  const yearString = params.year;
  const year = parseInt(yearString);
  const startDate = new Date(year, 0, 1); // January 1st
  const endDate = new Date(year, 11, 31); // December 31st
  const startDateString = startDate.toISOString().substring(0, 10); // "YYYY-MM-DD"
  const endDateString = endDate.toISOString().substring(0, 10); // "YYYY-MM-DD"

Which is then used as such:

      .gte("date", startDateString)
      .lte("date", endDateString);
